sessions/test_js_challenge.py

```diff
@@ -184,8 +177,23 @@ def test_get_challenge(self):
                "\r\n")
         self.client_expect_block(client, req)
 
+    def expect_restart(self, client, req, status_code, last_cookie):
```

Comment:
This is is the test for outdated cookie from a client, right? Please add a comment what the test verifies.